main/src/main/java/cgeo/geocaching/utils/FilterUtils.java

```diff
@@ -55,27 +68,33 @@ public static <T extends AbstractActivity & FilteredActivity> boolean openFilter
             final TextParam message = TextParam.concat(TextParam.id(R.string.cache_filter_storage_load_delete_nofilter_message),
                     TextParam.text(System.lineSeparator()),
                     TextParam.id(R.string.cache_filter_storage_clear_message));
-                SimpleDialog.of(filteredActivity).setTitle(R.string.cache_filter_storage_clear_title)
-                        .setPositiveButton(TextParam.id(R.string.cache_filter_storage_clear_button))
-                        .setMessage(message)
-                        .confirm(() ->
-                                filteredActivity.refreshWithFilter(GeocacheFilter.createEmpty(filterContext.get().isOpenInAdvancedMode()))
-                        );
-                return true;
-            }
-
-        final SimpleDialog.ItemSelectModel<GeocacheFilter> model = getGroupedFilterList(filters);
 
-        if (isFilterActive) {
-            SimpleDialog.of(filteredActivity).setTitle(R.string.cache_filter_storage_select_clear_title)
-                    .setNeutralButton(TextParam.id(R.string.cache_filter_storage_clear_button))
-                    .setNeutralAction(() ->
+            SimpleDialog.of(filteredActivity).setTitle(R.string.cache_filter_storage_clear_title)
+                    .setPositiveButton(TextParam.id(R.string.cache_filter_storage_clear_button))
+                    .setMessage(message)
+                    .confirm(() ->
                             filteredActivity.refreshWithFilter(GeocacheFilter.createEmpty(filterContext.get().isOpenInAdvancedMode()))
-                    ).selectSingle(model, filteredActivity::refreshWithFilter);
-        } else {
-            SimpleDialog.of(filteredActivity).setTitle(R.string.cache_filter_storage_select_title)
-                    .selectSingle(model, filteredActivity::refreshWithFilter);
+                    );
+            return true;
         }
+
+        final SimpleDialog.ItemSelectModel<GeocacheFilter> model = getGroupedFilterList(filters);
+
+        final Pair<TextParam, GeocacheFilter> lastFilterButton = getLastFilterActionForContext(filterContext);
+        SimpleDialog.of(filteredActivity)
+                .setTitle(isFilterActive ? TextParam.id(R.string.cache_filter_storage_select_clear_title) : TextParam.id(R.string.cache_filter_storage_select_title))
+                .setPositiveButton(null)
+                .setNeutralButton(isFilterActive ? TextParam.id(R.string.cache_filter_storage_clear_button) : null)
+                .setNegativeButton(lastFilterButton.first)
```

Comment:
I used the negative button for `Previous filter`, but to get it work, that both, negative and neutral buttons are shown (but no positive button), I had To get it work, I had to check for `buttonClickAction` in SimpleDialog.